new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java

```diff
@@ -731,6 +737,162 @@ public void testCreateNewCluster() {
+        + "\"value\": \"" + newWlSecretName + "\"},"
+        + "{\"op\": \"replace\", \"path\": \"/spec/introspectVersion\", "
+        + "\"value\": \"" + introspectVersion + "\"},"
+        + "{\"op\": \"add\", \"path\": \"/spec/restartVersion\", "
+        + "\"value\": \"" + newVersion + "\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}\n", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, introDomainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    //verify the introspector pod is created and runs
+    String introspectPodName = domainUid + "-" + "introspect-domain-job";
+
+    checkPodExists(introspectPodName, domainUid, introDomainNamespace);
+    checkPodDoesNotExist(introspectPodName, domainUid, introDomainNamespace);
+
+    //verify the pods are restarted
+    verifyRollingRestartOccurred(pods, 1, introDomainNamespace);
+
+    // verify the admin server service created
+    checkServiceExists(adminServerPodName, introDomainNamespace);
+
+    // verify admin server pod is ready
+    checkPodReady(adminServerPodName, domainUid, introDomainNamespace);
+
+    // verify new cluster managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Checking managed server service {0} is created in namespace {1}",
+          managedServerPodNamePrefix + i, introDomainNamespace);
+      checkServiceExists(managedServerPodNamePrefix + i, introDomainNamespace);
+    }
+
+    // verify new cluster managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Waiting for managed server pod {0} to be ready in namespace {1}",
+          managedServerPodNamePrefix + i, introDomainNamespace);
+      checkPodReady(managedServerPodNamePrefix + i, domainUid, introDomainNamespace);
+    }
+
+    logger.info("Getting the list of servers using the listServers");
+    String baseUri = "http://" + K8S_NODEPORT_HOST + ":" + adminServerT3Port + "/clusterview/";
+    String serverListUri = "ClusterViewServlet?listServers=true";
+    HttpResponse<String> response = null;
+    for (int i = 0; i < 5; i++) {
+      assertDoesNotThrow(() -> TimeUnit.SECONDS.sleep(10));
+      response = assertDoesNotThrow(() -> OracleHttpClient.get(baseUri + serverListUri, true));
+      assertEquals(200, response.statusCode(), "Status code not equals to 200");
+    }
+
+    // verify managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Checking {0} health", managedServerNameBase + i);
+      assertTrue(response.body().contains(managedServerNameBase + i + ":HEALTH_OK"),
+          "Didn't get " + managedServerNameBase + i + ":HEALTH_OK");
+    }
+
+    logger.info("Getting node port for default channel");
+    int serviceNodePort = assertDoesNotThrow(() -> getServiceNodePort(
```

Comment:
getServiceNodePort() doesn't throw Exception, check port returned is not -1.

Reply:
fixed